test/on_demand_operations_test.go

```diff
@@ -0,0 +1,393 @@
+							"operations-1-0": asdbv1.OperationPodRestart,
+							"operations-1-1": asdbv1.OperationPodRestart,
+						}
+
+						err = validateOperationTypes(ctx, aeroCluster, oldPodIDs, operationTypeMap)
+						Expect(err).ToNot(HaveOccurred())
+					},
+				)
+
+				It(
+					"Should execute operations on selected pods with dynamic config change", func() {
+						aeroCluster, err := getCluster(
+							k8sClient, ctx, clusterNamespacedName,
+						)
+						Expect(err).ToNot(HaveOccurred())
+
+						oldPodIDs, err := getPodIDs(ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						operations := []asdbv1.OperationSpec{
+							{
+								Kind:    asdbv1.OperationPodRestart,
+								PodList: []string{"operations-1-0"},
+							},
+						}
+
+						aeroCluster.Spec.EnableDynamicConfigUpdate = ptr.To(true)
+						aeroCluster.Spec.AerospikeConfig.Value["service"].(map[string]interface{})["proto-fd-max"] = 18000
+						aeroCluster.Spec.Operations = operations
+
+						err = updateCluster(k8sClient, ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						aeroCluster, err = getCluster(
+							k8sClient, ctx, clusterNamespacedName,
+						)
+						Expect(err).ToNot(HaveOccurred())
+
+						operationTypeMap := map[string]asdbv1.OperationKind{
+							"operations-1-0": asdbv1.OperationPodRestart,
+							"operations-1-1": "noRestart",
+						}
+
+						err = validateOperationTypes(ctx, aeroCluster, oldPodIDs, operationTypeMap)
+						Expect(err).ToNot(HaveOccurred())
+					},
+				)
+
+				It(
+					"Should execute podRestart operations on all pods along with scale down", func() {
+						aeroCluster, err := getCluster(
+							k8sClient, ctx, clusterNamespacedName,
+						)
+						Expect(err).ToNot(HaveOccurred())
+
+						aeroCluster.Spec.Size = 4
+
+						err = updateCluster(k8sClient, ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						aeroCluster, err = getCluster(
+							k8sClient, ctx, clusterNamespacedName,
+						)
+						Expect(err).ToNot(HaveOccurred())
+
+						oldPodIDs, err := getPodIDs(ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						operations := []asdbv1.OperationSpec{
+							{
+								Kind: asdbv1.OperationPodRestart,
+							},
+						}
+
+						aeroCluster.Spec.Operations = operations
+						aeroCluster.Spec.Size = 2
+
+						err = updateCluster(k8sClient, ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						aeroCluster, err = getCluster(
+							k8sClient, ctx, clusterNamespacedName,
+						)
+						Expect(err).ToNot(HaveOccurred())
+
+						operationTypeMap := map[string]asdbv1.OperationKind{
+							"operations-1-0": asdbv1.OperationPodRestart,
+							"operations-1-1": asdbv1.OperationPodRestart,
+						}
+
+						err = validateOperationTypes(ctx, aeroCluster, oldPodIDs, operationTypeMap)
+						Expect(err).ToNot(HaveOccurred())
+					},
+				)
+
+				It(
+					"Should execute podRestart operations on all pods along with upgrade", func() {
```

Comment:
Will upgrade plus podRestart lead to pod restart twice? One due the upgrade and second due to operation.
Also confirm, if the last rack reconcile flow will restart only Pod only once